azure-servicemanagement-legacy/tests/test_legacy_mgmt_misc.py

```diff
@@ -2166,7 +2166,7 @@ def test_list_os_images_public(self):
         self.assertIsNotNone(image.is_premium)
         self.assertIsNotNone(image.icon_uri)
         self.assertIsNotNone(image.privacy_uri)
-        self.assertGreaterEqual(len(image.recommended_vm_size), 0)
+        self.assertIsNotNone(image.recommended_vm_size)
```

Comment:
If this is just in the test code, couldn't we just make it so all of the mock/test OSImage objects have the field populated so the test passes reliably? Or is there a fear that that will touch too many things?

If that's the case, the test could only make a `len!=0` assertion for a given `image_name` where you know that mock image has the field populated and thus should be present after deserialization?

(Not sure if that helps, I don't 100% grok the issue or how the tests are setup....)